e2e_testing/README.md

```diff
@@ -1 +1,87 @@
+
+The tests then should interpret the rendered UI as would a user or an assistive technology. Typically source code would be annotated with HTML attributes for the tests to select on, or tests would make use of DOM XPaths or CSS selector paths. These approaches should be avoided as they incur a reliance on the implementation detail, can be brittle and do not reflect a user's view of the system. Playwright's best practices include a recommendation to [test user-visible behavior](https://playwright.dev/docs/best-practices#test-user-visible-behavior) and an element [Locator](https://playwright.dev/docs/locators) preference to select on accessibility attributes and textual content. Where practical, this approach should be followed.
+
+## Development
+
+Playwright provides a [UI workbench](https://playwright.dev/docs/test-ui-mode) for running and debugging tests during development. This can be started from the `e2e_testing` directory with:
+
+```bash
+yarn start
+```
+
+For headless testing from the command line and on CI, we can run the tests with:
+
+```bash
+yarn test
+```
+
+To target specific environments, set `BASE_URL` on the environment, e.g:
+
+```bash
+BASE_URL=https://mitopen.odl.mit.edu/ yarn test
+```
+
+NPM scripts are provided for our RC and Production environments:
+
+```bash
+yarn test:rc
+yarn test:production
+```
+
+A Docker Compose configuration at [../docker-compose-e2e-tests.yml](../docker-compose-e2e-tests.yml) provides the minimal run configuration for starting services needed for testing, the Postgres database, the backend web service and our nginx reverse proxy:
+
+Services can be started with:
+
+```bash
+docker compose -f docker-compose-e2e-tests.yml up web db nginx
+```
+
+The tests themselves are also containerized so the following command will start all services, run the tests and shut down services when the tests have completed:
+
+```bash
+docker compose -f docker-compose-e2e-tests.yml up --exit-code-from e2e-tests
+```
+
+## Data Handling
+
+Applying and tearing down data adds significant overhead to writing tests. This can be minimized by writing SQL alongside any tests that are expecting specific data, that can be applied to a fresh ephemeral database locally and in CI that is disposed of after a test run. This also guarantees that tests are dependent only on the codebase and not any data that happens to have been set during an instances lifetime on a hosted environment. We have the added benefit that we do not need to write teardown code to leave the database in its original state.
+
+These test fixtures can be written in plain SQL and any files in the adjacent [./fixtures](./fixtures) directory will be applied to the database whilst standing up services in CI. As a suggestion, SQL inserts can be written in [JSON format](https://www.postgresql.org/docs/9.6/functions-json.html) for readability.
```

Comment:
nit: If we keep postgres docs, i think we should link to current version...probably even if it isn't quite the version we use.
```suggestion
These test fixtures can be written in plain SQL and any files in the adjacent [./fixtures](./fixtures) directory will be applied to the database whilst standing up services in CI. As a suggestion, SQL inserts can be written in [JSON format](https://www.postgresql.org/docs/9.6/functions-json.html) for readability.
```